evalml/tests/model_understanding_tests/test_graphs.py

```diff
@@ -1418,3 +1420,76 @@ def test_visualize_decision_trees(fitted_tree_estimators, tmpdir):
     assert src.format == 'pdf'
     assert isinstance(src, graphviz.Source)
+
+
+@pytest.mark.parametrize("n_components", [2.0, -2, 0])
+def test_t_sne_errors_n_components(n_components):
+    X = np.array([[0, 0, 0], [0, 1, 1], [1, 0, 1], [1, 1, 1]])
+
+    with pytest.raises(ValueError, match=f"The parameter n_components must be of type integer and greater than 0"):
+        t_sne(X, n_components=n_components)
+
+
+@pytest.mark.parametrize("perplexity", [-2, -1.2])
+def test_t_sne_errors_perplexity(perplexity):
+    X = np.array([[0, 0, 0], [0, 1, 1], [1, 0, 1], [1, 1, 1]])
+
+    with pytest.raises(ValueError, match=f"The parameter perplexity must be non-negative"):
+        t_sne(X, perplexity=perplexity)
+
+
+@pytest.mark.parametrize("data_type", ['np', 'pd', 'ww'])
+def test_t_sne(data_type):
+    if data_type == 'np':
+        X = np.array([[0, 0, 0], [0, 1, 1], [1, 0, 1], [1, 1, 1]])
+    elif data_type == 'pd':
+        X = pd.DataFrame([[0, 0, 0], [0, 1, 1], [1, 0, 1], [1, 1, 1]])
+    elif data_type == 'ww':
+        X = ww.DataTable(np.array([[0, 0, 0], [0, 1, 1], [1, 0, 1], [1, 1, 1]]))
+
+    output_ = t_sne(X, n_components=2, perplexity=50, learning_rate=200.0)
+    assert isinstance(output_, np.ndarray)
+
+
+@pytest.mark.parametrize("marker_line_width", [-2, -1.2])
+def test_t_sne_errors_marker_line_width(marker_line_width, has_minimal_dependencies):
+    if has_minimal_dependencies:
+        pytest.skip("Skipping plotting test because plotly not installed")
+    X = np.array([[0, 0, 0], [0, 1, 1], [1, 0, 1], [1, 1, 1]])
+
+    with pytest.raises(ValueError, match=f"The parameter marker_line_width must be non-negative"):
+        graph_t_sne(X, marker_line_width=marker_line_width)
+
+
+@pytest.mark.parametrize("marker_size", [-2, -1.2])
+def test_t_sne_errors_marker_size(marker_size, has_minimal_dependencies):
+    if has_minimal_dependencies:
+        pytest.skip("Skipping plotting test because plotly not installed")
+    X = np.array([[0, 0, 0], [0, 1, 1], [1, 0, 1], [1, 1, 1]])
+
+    with pytest.raises(ValueError, match=f"The parameter marker_size must be non-negative"):
+        graph_t_sne(X, marker_size=marker_size)
+
+
+@pytest.mark.parametrize("data_type", ['np', 'pd', 'ww'])
+@pytest.mark.parametrize("perplexity", [0, 4.6, 100])
+@pytest.mark.parametrize("learning_rate", [100.0, -15, 0])
+def test_graph_t_sne(data_type, perplexity, learning_rate):
+    go = pytest.importorskip('plotly.graph_objects', reason='Skipping plotting test because plotly not installed')
+    if data_type == 'np':
+        X = np.array([[0, 0, 0], [0, 1, 1], [1, 0, 1], [1, 1, 1]])
+    elif data_type == 'pd':
+        X = pd.DataFrame([[0, 0, 0], [0, 1, 1], [1, 0, 1], [1, 1, 1]])
+    elif data_type == 'ww':
+        X = np.array([[0, 0, 0], [0, 1, 1], [1, 0, 1], [1, 1, 1]])
+        X = _convert_to_woodwork_structure(X)
+        X = _convert_woodwork_types_wrapper(X.to_dataframe())
```

Comment:
This is converting back to dataframe so we're not covering the data table case.

Reply:
Good catch